substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/analysis/Inflation.java

```diff
@@ -24,718 +24,28 @@
-                /* If we hit JDK-7183985 just encode the exception. */
-                hub.setAnnotationsEncoding(e);
-            }
-
-            /*
-             * Support for Java enumerations.
-             */
-            if (type.isEnum() && hub.shouldInitEnumConstants()) {
-                if (getHostVM().getClassInitializationSupport().shouldInitializeAtRuntime(type)) {
-                    hub.initEnumConstantsAtRuntime(type.getJavaClass());
-                } else {
-                    /*
-                     * We want to retrieve the enum constant array that is maintained as a private
-                     * static field in the enumeration class. We do not want a copy because that
-                     * would mean we have the array twice in the native image: as the static field,
-                     * and in the enumConstant field of DynamicHub. The only way to get the original
-                     * value is via a reflective field access, and we even have to guess the field
-                     * name.
-                     */
-                    AnalysisField found = null;
-                    for (AnalysisField f : type.getStaticFields()) {
-                        if (f.getName().endsWith("$VALUES")) {
-                            if (found != null) {
-                                /*
-                                 * Enumeration has more than one static field with enumeration
-                                 * values. Bailout and use Class.getEnumConstants() to get the value
-                                 * instead.
-                                 */
-                                found = null;
-                                break;
-                            }
-                            found = f;
-                        }
-                    }
-                    Enum<?>[] enumConstants;
-                    if (found == null) {
-                        /*
-                         * We could not find a unique $VALUES field, so we use the value returned by
-                         * Class.getEnumConstants(). This is not ideal since
-                         * Class.getEnumConstants() returns a copy of the array, so we will have two
-                         * arrays with the same content in the image heap, but it is better than
-                         * failing image generation.
-                         */
-                        enumConstants = (Enum<?>[]) type.getJavaClass().getEnumConstants();
-                    } else {
-                        enumConstants = (Enum<?>[]) SubstrateObjectConstant.asObject(getConstantReflectionProvider().readFieldValue(found, null));
-                        assert enumConstants != null;
-                    }
-                    hub.initEnumConstants(enumConstants);
-                }
-            }
-        }
-    }
+/**
+ * Extension of StaticAnalysisEngine providing access to more svm specific methods.
+ */
+public interface Inflation extends BigBang {
```

Comment:
`Inflation` is an interface, whereas `NativeImagePointsToAnalysis ` is an actual implementation of it. Since there will be a different implementation of `BigBang`for the Reachability Analysis, I'm keeping the Inflation interface for that class (or its subclass) to implement as well. There are places where `Inflation` is used, not `BigBang` and once there are multiple implementations of it, we would have to reintroduce it anyway, if I am not mistaken.